### PR TITLE
abstract test sequencer run & raise error when accessing identifier before eval

### PR DESCRIPTION
**Pull request by izi-on**

### CODE REVIEW

File: captain/parser/bool_parser/expressions/exceptions.py [hunk 1]
```diff
@@ -44,3 +44,9 @@ class InvalidIdentifier(Exception):
     def __init__(self, message=""):
         self.message = "Invalid identifier: " + message
         super().__init__(self.message)
+
+
+class EarlyIdentifier(Exception):
+    def __init__(self, message=""):
+        self.message = f"Cannot access identifier {message} before evaluation"
```
Comment (LatentDream) [inline captain/parser/bool_parser/expressions/exceptions.py:51]: The error is good, but a little bit too "comp sci" for a user. Maybe something like:
```
f"Unable to access test result for {message} as it's executed afterward"
```

File: captain/utils/test_sequencer/run_test_sequence.py [hunk 2]
```diff
@@ -101,56 +108,96 @@ async def _stream_result_to_frontend(
...
+                {
+                    "if": (None, _case_if_node),
+                },
+            ),
+        },
+    )
+    matcher, cur = map_to_handler
+    while not callable(cur):
+        matcher, cur = cur[node.__dict__[matcher]]
+    children_getter, test_result = await cur(node)  # sus name for the variable
```
Comment (LatentDream) [inline captain/utils/test_sequencer/run_test_sequence.py:175]: Ahahha, would remove the comment tho

**Review by LatentDream — CHANGES_REQUESTED** [review]
lgtm, just some small things.
Also not sure we need a __init__ in each directory ?

**Review by itsjoeoui — APPROVED** [review]
lgtm

**Pull request merged**